xxhr/util.hpp

```diff
@@ -168,11 +168,15 @@ namespace util {
   inline std::string decode64(const std::string &val) {
       using namespace boost::archive::iterators;
       using It = transform_width<binary_from_base64<std::string::const_iterator>, 8, 6>;
-      return boost::algorithm::trim_right_copy_if(std::string(It(std::begin(val)), It(std::end(val))), [](char c) {
-          return c == '\0';
-      });
+      // See https://svn.boost.org/trac10/ticket/5629#comment:9
+      // Boost binary_from_base64 transforms '=' into '\0', they need to be removed to support binary data
+      auto decoded_with_zeroed_padding = std::string(It(std::begin(val)), It(std::end(val)));
+      auto padding_count = std::count(val.end() - std::min(std::size_t{2}, val.size()), val.end() , '=');
+      auto decoded = decoded_with_zeroed_padding.substr(0,decoded_with_zeroed_padding.size()-padding_count);
+      return decoded;
```

Comment:
The code above makes me cringe, but I see why you solved it this way.

Reply:
Note from our call : count before transform_width It, to avoid creating 2 copies and simplify code. Thanks @pysco68

Reply:
Done